tests/components/ActionButton/StatefulActionButton.spec.ts

```diff
@@ -0,0 +1,241 @@
+            },
+            {
+                describe: 'if initialState is not defined',
+                beforeEach: () => {
+                    testSubject = new StatefulActionButton(document.createElement('div'), { states: [initialState], initialState: undefined });
+                },
+                expectedWarningMessage: 'The stateful action button cannot render if the initial states is not defined.',
+            },
+            {
+                describe: 'if states does not include initialState',
+                beforeEach: () => {
+                    testSubject = new StatefulActionButton(document.createElement('div'), {
+                        states: [createSpiedState('notInitialState')[0]],
+                        initialState: initialState,
+                    });
+                },
+                expectedWarningMessage: 'The stateful action button cannot render if the initial state is not in the list of states.',
+            },
+            {
+                describe: 'if first allowedTransitions contains a transition with a origin state not included in options.state',
+                beforeEach: () => {
+                    testSubject = new StatefulActionButton(document.createElement('div'), {
+                        states: [initialState],
+                        initialState: initialState,
+                        allowedTransitions: [{ from: createSpiedState('someState')[0], to: initialState }],
+                    });
+                },
+                expectedWarningMessage:
+                    'The stateful action button cannot render if one of its transition used a state that is not in the list of states:\n\tOrigin of Transition #0 is not in the list of states. Ensure to use the same object reference as in the options.states.',
+            },
+            {
+                describe: 'if second allowedTransitions contains a transition with a origin state not included in options.state',
+                beforeEach: () => {
+                    testSubject = new StatefulActionButton(document.createElement('div'), {
+                        states: [initialState],
+                        initialState: initialState,
+                        allowedTransitions: [
+                            { from: initialState, to: initialState },
+                            { from: createSpiedState('someState')[0], to: initialState },
+                        ],
+                    });
+                },
+                expectedWarningMessage:
+                    'The stateful action button cannot render if one of its transition used a state that is not in the list of states:\n\tOrigin of Transition #1 is not in the list of states. Ensure to use the same object reference as in the options.states.',
+            },
+            {
+                describe: 'if first allowedTransitions contains a transition with a destination state not included in options.state',
```

Comment:
Same thing as my previous comment. Unless I misunderstood the test case, I would suggest to setup two transitions and make the last one invalid. The test case should be renamed

    if allowedTransitions contains a transition with a destination state not included in options.state